src/main/java/org/terasology/taskSystem/nui/TaskSelectionScreenLayer.java

```diff
@@ -20,90 +20,87 @@
-        gymButton = find(Constants.GYM_BUTTON_ID, UIButton.class);
-        classroomButton = find(Constants.CLASSROOM_BUTTON_ID, UIButton.class);
-        storageButton = find(Constants.STORAGE_BUTTON_ID, UIButton.class);
-        laboratoryButton = find(Constants.LABORATORY_BUTTON_ID, UIButton.class);
-
         cancelButton = find(Constants.CANCEL_BUTTON_ID, UIButton.class);
 
-        plantCommand.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Plant);
-        });
-
-        guardCommand.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Guard);
-        });
-
-        hospitalButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Hospital);
-        });
-
-        dinerButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Diner);
-        });
-
-        gymButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Gym);
-        });
-
-        classroomButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Classroom);
-        });
-
-        storageButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Storage);
-        });
-
-        laboratoryButton.subscribe(button -> {
-            sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.Laboratory);
-        });
+        populateTasksList();
+        populateBuildingsList();
 
         cancelButton.subscribe(button -> {
             sendSetTaskTypeEvent();
         });
     }
 
-    public void sendSetTaskTypeEvent () {
+    private void sendSetTaskTypeEvent () {
         localPlayer.getCharacterEntity().send(new SetTaskTypeEvent());
     }
 
-    public void sendSetTaskTypeEvent(String assignedTaskType) {
+    private void sendSetTaskTypeEvent(String assignedTaskType) {
         localPlayer.getCharacterEntity().send(new SetTaskTypeEvent(assignedTaskType));
     }
 
-    public void sendSetTaskTypeEvent (String assignedTaskType, BuildingType buildingType) {
+    private void sendSetTaskTypeEvent (String assignedTaskType, BuildingType buildingType) {
         localPlayer.getCharacterEntity().send(new SetTaskTypeEvent(assignedTaskType, buildingType));
     }
+
+    private void populateBuildingsList() {
+        buildingsList = find(Constants.BUILDINGS_LIST_ID, UIList.class);
+        buildingsList.subscribeSelection(new ItemSelectEventListener<String>() {
+            @Override
+            public void onItemSelected(UIWidget widget, String item) {
+                sendSetTaskTypeEvent(AssignedTaskType.Build, BuildingType.valueOf(item));
+            }
+        });
+
+        List<String> buttonList = new ArrayList<>();
+        buttonList.add("Diner");
+        buttonList.add("Storage");
+        buttonList.add("Laboratory");
+        buttonList.add("Hospital");
+        buildingsList.setList(buttonList);
```

Comment:
Likewise to how you can avoid defining the values in the selection code, you can automatically get all the values in an Enum, and specify a human readable formay for each entry as well.